### PR TITLE
support-replay-packet

### PR DESCRIPTION
**Pull request by ShakoHo**
@askeing @ypwalter r?

### CODE REVIEW

File: runtest.py [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
   runtest.py (-h | --help)
 
 Options:
   -h --help                 Show this screen.
   --max-run=<int>           Test run max no [default: 40].
   --max-retry=<int>         Test failed retry max no [default: 15].
-  --keep-browser=<int>      Keep the browser open after test script executed [default: 0]
-  --profiler=<str>          Enabled profiler, current support profiler:avconv,geckoprofiler,harexport,chrometracing,fxall,justprofiler [default: avconv]
+  --keep-browser            Keep the browser open after test script executed
+  --profiler=<str>          Enabled profiler, current support profiler:avconv,geckoprofiler,harexport,chrometracing,fxall,justprofiler, mitmdump [default: avconv]
```
Comment (askeing) [inline runtest.py]: remove space for for consistent format

File: lib/profiler/mitmDumpProfiler.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        else:
+            replay_cmd_list = ["mitmdump", "--no-http2", "-S", flow_file_fp, "--no-pop", "--norefresh"]
+            self.process = subprocess.Popen(replay_cmd_list)
+
+    def stop_recording(self, **kwargs):
+        if platform.system().lower() == "windows":
+            subprocess.Popen("taskkill /IM mitmdump.exe /T /F", shell=True)
+        else:
+            self.process.send_signal(3)
+
```
Comment (ypwalter) [inline lib/profiler/mitmDumpProfiler.py]: empty line at end of file

File: runtest.py [hunk 3]
```diff
@@ -53,9 +54,10 @@ def kill_legacy_process(self):
     def get_test_env(self, **kwargs):
         result = os.environ.copy()
         result['PROFILER'] = self.profiler
-        result['KEEP_BROWSER'] = str(self.keep_browser)
-        result['ENABLE_ONLINE'] = str(self.online)
+        result['KEEP_BROWSER'] = str(int(self.keep_browser))
+        result['ENABLE_ONLINE'] = str(int(self.online))
```
Comment (askeing) [inline runtest.py]: hmm, I think we should do something for `--keep-browser` and `--online` switches.
let's modify them in the future refactoring work :)

Comment (ShakoHo) [inline runtest.py]: agree ...XD

File: lib/profiler/mitmDumpProfiler.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+class MitmDumpProfiler(BaseProfiler):
+
+    process = None
+
+    def start_recording(self):
+        default_dump_flow_cmd = ["mitmdump", "--no-http2", "-w"]
+        flow_file_fp = os.path.join(self.env.DEFAULT_FLOW_DIR, self.env.flow_file_fp)
+        if not os.path.exists(self.env.DEFAULT_FLOW_DIR):
+            os.mkdir(self.env.DEFAULT_FLOW_DIR)
```
Comment (askeing) [inline lib/profiler/mitmDumpProfiler.py]: just like what we discuss before, I'll suggest using `mkdir -p`-liked method.
And now, I found it, `os.makedirs()`, from https://docs.python.org/2/library/os.html#os.makedirs :P

**Comment by askeing** [comment]
LGTM :)

**Pull request merged**